circle.yml

```diff
@@ -1,16 +1,28 @@
 machine:
-  ghc:
-    version: 7.10.1
+  pre:
+    # https://github.com/commercialhaskell/stack/issues/1658
+    - sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.6 20
+    - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.6 20
+    - sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.9 10
+    - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.9 10
 
 dependencies:
+  pre:
+    - wget https://github.com/commercialhaskell/stack/releases/download/v0.1.6.0/stack-0.1.6.0-linux-x86_64.tar.gz -O /tmp/stack.tar.gz
+    - tar xvzOf /tmp/stack.tar.gz stack-0.1.6.0-linux-x86_64/stack > /tmp/stack
+    - chmod +x /tmp/stack && sudo mv /tmp/stack /usr/bin/stack
   override:
+    - stack setup
     - ./bin/setup
+  cache_directories:
+    - "~/.stack"
 
 test:
   override:
-    - cabal run --  clean
-    - cabal run --  build
-    - cabal run --  check --internal-links
+    - stack build
+    - stack exec site -- clean
```

Comment:
Probably don't need this. Do you expect `_site` to have content in a fresh checkout?